pennylane/data/dataset.py

```diff
@@ -208,14 +217,22 @@ def __get_attribute_from_filename(self, filename):
 
     def __getattribute__(self, name):
         try:
-            return super().__getattribute__(name)
+            val = super().__getattribute__(name)
+            if val is None and self._is_standard and name in self.attrs:
```

Comment:
one of the scariest lines of code I've written. I originally had `self._is_standard` as the first condition, but that caused an infinite loop. Only because I KNOW that it won't be None is this safe. The danger here is that `self._is_standard` and `self.attrs` causes another call to `self.__getattribute__`. It should be fine because `self._is_standard` is _always_ either True or False, and `self.attrs` is always a dict (I think `vars(self)` in that property triggers another `__getattribute__` for `__dict__`, but that is non-None as well). Because none of those are `None`, their internal call's `if` will eagerly evaluate to False, and it will carry on to return their respective value.

Reply:
Is there still any worst case scenario that this might still lead to an infinite loop?

Reply:
maybe 😄 idk lol. if none of our unittests hit it, it feels like it'll be hard for the user to hit. if they do, I hope they file a bug! Python can tell if a function calls itself recursively more than some number of times with the exact same arguments, and raises an error if it happens. The user will know very quickly.

Reply:
Scary.